hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/ReconUtils.java

```diff
@@ -246,25 +257,106 @@ public void untarCheckpointFile(File tarFile, Path destPath)
     }
   }
 
+
+  /**
+   * Constructs the full path of a key from its OmKeyInfo using a bottom-up approach, starting from the leaf node.
+   *
+   * The method begins with the leaf node (the key itself) and recursively prepends parent directory names, fetched
+   * via NSSummary objects, until reaching the parent bucket (parentId is -1). It effectively builds the path from
+   * bottom to top, finally prepending the volume and bucket names to complete the full path. If the directory structure
+   * is currently being rebuilt (indicated by the isRebuilding flag), this method returns an empty string to signify
+   * that path construction is temporarily unavailable.
+   *
+   * @param omKeyInfo The OmKeyInfo object for the key
+   * @return The constructed full path of the key as a String, or an empty string if a rebuild is in progress and
+   *         the path cannot be constructed at this time.
+   * @throws IOException
+   */
+
+  public static String constructFullPath(OmKeyInfo omKeyInfo,
+                                         ReconNamespaceSummaryManager reconNamespaceSummaryManager,
+                                         ReconOMMetadataManager omMetadataManager)
+      throws IOException {
+
+    // Return empty string to signify that path construction is temporarily unavailable
+    if (isRebuilding) {
+      return "";
+    }
+
+    StringBuilder fullPath = new StringBuilder(omKeyInfo.getKeyName());
+    long parentId = omKeyInfo.getParentObjectID();
+    boolean isDirectoryPresent = false;
+
+    while (parentId != 0) {
+      NSSummary nsSummary = reconNamespaceSummaryManager.getNSSummary(parentId);
+      if (nsSummary == null) {
+        break;
+      }
+      if (nsSummary.getParentId() == -1 && !rebuildTriggered) {
+        synchronized (ReconUtils.class) {
```

Comment:
Thanks for the comment @sumitagrawl
We'll replace the volatile boolean with an AtomicBoolean for rebuildTriggered. Since it's to be triggered only once in the lifetime of the process, we won't reset this value after setting it.